src/Middleware/OutputCaching/test/OutputCachePolicyProviderTests.cs

```diff
@@ -0,0 +1,410 @@
+    [InlineData(StatusCodes.Status414RequestUriTooLong)]
+    [InlineData(StatusCodes.Status415UnsupportedMediaType)]
+    [InlineData(StatusCodes.Status416RequestedRangeNotSatisfiable)]
+    [InlineData(StatusCodes.Status417ExpectationFailed)]
+    [InlineData(StatusCodes.Status418ImATeapot)]
+    [InlineData(StatusCodes.Status419AuthenticationTimeout)]
+    [InlineData(StatusCodes.Status421MisdirectedRequest)]
+    [InlineData(StatusCodes.Status422UnprocessableEntity)]
+    [InlineData(StatusCodes.Status423Locked)]
+    [InlineData(StatusCodes.Status424FailedDependency)]
+    [InlineData(StatusCodes.Status426UpgradeRequired)]
+    [InlineData(StatusCodes.Status428PreconditionRequired)]
+    [InlineData(StatusCodes.Status429TooManyRequests)]
+    [InlineData(StatusCodes.Status431RequestHeaderFieldsTooLarge)]
+    [InlineData(StatusCodes.Status451UnavailableForLegalReasons)]
+    [InlineData(StatusCodes.Status500InternalServerError)]
+    [InlineData(StatusCodes.Status501NotImplemented)]
+    [InlineData(StatusCodes.Status502BadGateway)]
+    [InlineData(StatusCodes.Status503ServiceUnavailable)]
+    [InlineData(StatusCodes.Status504GatewayTimeout)]
+    [InlineData(StatusCodes.Status505HttpVersionNotsupported)]
+    [InlineData(StatusCodes.Status506VariantAlsoNegotiates)]
+    [InlineData(StatusCodes.Status507InsufficientStorage)]
+    [InlineData(StatusCodes.Status508LoopDetected)]
+    [InlineData(StatusCodes.Status510NotExtended)]
+    [InlineData(StatusCodes.Status511NetworkAuthenticationRequired)]
+    public async Task IsResponseCacheable_NonSuccessStatusCodes_NotAllowed(int statusCode)
+    {
+        var sink = new TestSink();
+        var context = TestUtils.CreateTestContext(sink);
+        context.HttpContext.Response.StatusCode = statusCode;
+
+        var policy = new OutputCachePolicyBuilder().Build();
+        await policy.ServeResponseAsync(context);
+
+        Assert.True(context.AllowCacheLookup);
+        Assert.False(context.AllowCacheStorage);
+        TestUtils.AssertLoggedMessages(
+            sink.Writes,
+            LoggedMessage.ResponseWithUnsuccessfulStatusCodeNotCacheable);
+    }
+
+    [Fact]
+    public async Task IsResponseCacheable_NoExpiryRequirements_IsAllowed()
+    {
+        var sink = new TestSink();
+        var context = TestUtils.CreateTestContext(sink);
+        context.HttpContext.Response.StatusCode = StatusCodes.Status200OK;
+
+        var utcNow = DateTimeOffset.UtcNow;
+        context.HttpContext.Response.Headers.Date = HeaderUtilities.FormatDate(utcNow);
+        context.ResponseTime = DateTimeOffset.MaxValue;
+
+        var policy = new OutputCachePolicyBuilder().Build();
+        await policy.ServeResponseAsync(context);
+
+        Assert.True(context.AllowCacheStorage);
+        Assert.True(context.AllowCacheLookup);
+        Assert.Empty(sink.Writes);
+    }
+
+    [Fact]
+    public async Task IsResponseCacheable_MaxAgeOverridesExpiry_ToAllowed()
```

Comment:
Why are this and the one below "ToAllowed" instead of IsAllowed or Allowed? Just checking if it's a typo or there's something different about these that I'm missing.